crates/neon/src/types_impl/promise.rs

```diff
@@ -49,9 +49,74 @@ const BOUNDARY: FailureBoundary = FailureBoundary {
     feature = "promise-api",
     deprecated = "`promise-api` feature has no impact and may be removed"
 )]
-/// The JavaScript [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) value.
+/// The type of JavaScript
+/// [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+/// objects.
 ///
-/// [`JsPromise`] may be constructed with [`Context::promise`].
+/// [`JsPromise`] instances may be constructed with [`Context::promise`].
+///
+/// # Example
+///
+/// ```
+/// # use neon::prelude::*;
+/// fn resolve_promise(mut cx: FunctionContext) -> JsResult<JsPromise> {
+///     let (deferred, promise) = cx.promise();
+///     let msg = cx.string("Hello, World!");
+///
+///     deferred.resolve(&mut cx, msg);
+///
+///     Ok(promise)
+/// }
+/// ```
+///
+/// # Example: Asynchronous task
+///
+/// This example uses the [linkify](https://crates.io/crates/linkify) crate in an
+/// asynchronous task, i.e. a
+/// [Node worker pool](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/)
+/// thread, to find all the links in a text string.
+///
+/// More robust implementations might use a custom Rust thread or thread pool to avoid
```

Comment:
I might remove "more robust" since it's use case specific whether it's better to use a Rust thread or the libuv thread. Maybe say, "alternate implementations."